UI/src/view/linear_asset/serviceRoadStyle.js

```diff
@@ -21,6 +21,21 @@
       return  me.lineFeatures(me.getNewFeatureProperties(linearAssets)).concat(me.renderOverlays(linearAssets));
     };
 
+    this.getNewFeatureProperties = function(linearAssets){
+      var linearAssetsWithType = _.map(linearAssets, function(linearAsset) {
+        var expired = _.isUndefined(linearAsset.value);
+        var type =  me.isUnknown(linearAsset) ? { type: 'unknown' } : {type: 'line'};
+        return _.merge({}, linearAsset, { expired: expired }, type);
+      });
+      var offsetBySideCode = function(linearAsset) {
+        return GeometryUtils.offsetBySideCode(applicationModel.zoom.level, linearAsset);
+      };
+      var linearAssetsWithAdjustments = _.map(linearAssetsWithType, offsetBySideCode);
+      return _.sortBy(linearAssetsWithAdjustments, function(asset) {
+        return asset.expired ? -1 : 1;
+      });
+    };
```

Comment:
Is there any reason for override this method getNewFeatureProperties instead of replace 'experied' to 'hasvalue' in ServiceRoadStyle?

Reply:
Using the one on the base class and changing the rules to 'hasAsset' onServiceRoadStyle also changes the way the overlay lines are drawn. From my experience it is not worth to start changing everything, and better to just override it here.